core/eolearn/core/utilities.py

```diff
@@ -475,3 +476,80 @@ def constant_pad(X, multiple_of, up_down_rule='even', left_right_rule='even', pa
 def bgr_to_rgb(bgr):
     """Converts Blue, Green, Red to Red, Green, Blue."""
     return bgr[..., [2, 1, 0]]
+
+def map_image_slices(data, func2d):
+    """Iterate over time and band dimensions and apply a function to each slice.
+
+    Returns a new array with the combined results.
+    
+    :param data: input array
+    :type data: array of shape (timestamps, rows, columns, channels)
+    :param func2d: Mapping function that is applied on each 2d image slice. All outputs must have the same shape.
+    :type func2d: function (rows, columns) -> (new_rows, new_columns)
+    """
+    
+    idx = np.ndindex(data.shape[0], data.shape[3])
+    output = None
+    for i,j in idx:
+        res = func2d(data[i,...,j])
+        if output is None:
+            output = np.zeros((data.shape[0],) + res.shape + (data.shape[3],), dtype=res.dtype)
+        
+        output[i,...,j] = res
+    
+    return output
+
+def resize_images(data, new_size=None, scale_factors=None, anti_alias=True, interpolation='linear'):
+    """Resizes the images acording to given size or scale factors.
+
+    To specify the new scale use one of `new_size` or `scale_factors` parameters.
+    
+    :param data: input array
+    :type data: array of shape (timestamps, rows, columns, channels)
+    :param new_size: New size of the data (width, height)
+    :type new_size: (int, int)
```

Comment:
what happens if new_size is not int?

Reply:
Currently the error is raised by OpenCV. (`TypeError: integer argument expected, got float`)

There are other options. We could raise our own error, or just round to the nearest integer?

Reply:
i would suggest raising an error

Reply:
Done